docs/app/pages/blog/posts/2017-08-24-Protractor-Testing.md

```diff
@@ -0,0 +1,189 @@
+End-to-end (E2E) testing complements unit testing and does not replace it. It allows you to test the way users see your application.
+
+Protractor is an end-to-end, open source test framework for Angular. It uses the Jasmine framework by default and runs on top of Selenium WebDriver. Tests may be run against a local Selenium server or against a remote server e.g. a Selenium Grid system. Tests may be written in JavaScript and TypeScript.
+
+The UXAspects project includes a suite of E2E tests in the 'e2e' folder of the project repository.
+
+### Installation
+
+Install [Node.js](https://nodejs.org/), which provides the npm package manager.
+
+Execute the command
+```
+npm install -g grunt-cli
+```
+to install the grunt command line interface.
+
+Follow the instructions in the "Building from Source" section in the [Getting Started](https://uxaspects.github.io/UXAspects/#/gettingstarted) page of the UX Aspects documentation to build the components to be tested.
+
+Before the first test run, execute
+```
+./node_modules/.bin/webdriver-manager update
+```
+to download the binaries necessary to get an instance of the Selenium Server running.
+
+Execute the command
+```
+grunt e2e
+```
+to run the tests.
+
+
+### Architecture
+
+All Protractor methods are asynchronous and return promises. WebDriver maintains a queue of pending promises called the control flow. Promises are queued and resolved in the order they were created in.
+
+In our tests, elements are located on the test page by using ‘by.ID’ and ‘by.CSS’ locators.
+
+The Protractor global elements() function takes a locator for an element and returns that element. When using CSS selectors as a locator, the shortcut $() notation may be used. For example, 
+```
+checkbox1.$('div .ux-checked')
+```
+ is the same as 
+```
+checkbox1.element(by.css('div .ux-checked'))
+```.
```

Comment:
I think the full stop here is preventing the code snippet from closing as code snippets after aren't highlighted but the regular paragraphs are (in GitHub markdown viewer at least, however it should display well in both GitHub and blog page). 
If this is all supposed to be on one line surround with single ticks instead.